src/Microsoft.OpenApi/Validations/Rules/RuleHelpers.cs

```diff
@@ -52,6 +53,15 @@ public static bool IsEmailAddress(this string input)
                 return;
             }
 
+            // Resolve the Json schema in memory before validating the data types.
+            var reference = schema.GetRef();
+            if (reference != null)
+            {
+                var referencePath = string.Concat("https://registry", reference.OriginalString.Split('#').Last());
+                var resolvedSchema = (JsonSchema)SchemaRegistry.Global.Get(new Uri(referencePath));
```

Comment:
I'm assuming this reference to the global registry goes away with Irvine's changes?

Reply:
Yes this goes away in Irvine's PR